Not strictly necessary as error aborts the build, the difference being
though is that ERROR just fails as it is not understood, while 'error'
displays the message that follows it.

Originally I added this a Makefile action:

COMPONENT_PREP_ACTION +=  $(FIND) $(@D) -type f -name \*.\[ch\] -exec \
			$(GSED) -i -e 's/\(\#[ ]*\)ERROR /\1error /' {} +;

Though thought it was best to create this patch file,

find . -type f -name \*.\[ch\] -exec gsed -i -e 's/\(\#[ ]*\)ERROR /\1error /' {} +

I shall followup with upstream and ask them to use error rather than ERROR.

diff -r 912b8d5daca2 -r 7ce8b17603d2 libmilter/engine.c
--- a/libmilter/engine.c	Thu Mar 09 08:32:49 2023 -0800
+++ b/libmilter/engine.c	Thu Mar 09 08:40:07 2023 -0800
@@ -64,28 +64,28 @@
 #define CI_EOH		6
 #define CI_LAST		CI_EOH
 #if CI_LAST < CI_DATA
-# ERROR "do not compile with CI_LAST < CI_DATA"
+# error "do not compile with CI_LAST < CI_DATA"
 #endif
 #if CI_LAST < CI_EOM
-# ERROR "do not compile with CI_LAST < CI_EOM"
+# error "do not compile with CI_LAST < CI_EOM"
 #endif
 #if CI_LAST < CI_EOH
-# ERROR "do not compile with CI_LAST < CI_EOH"
+# error "do not compile with CI_LAST < CI_EOH"
 #endif
 #if CI_LAST < CI_RCPT
-# ERROR "do not compile with CI_LAST < CI_RCPT"
+# error "do not compile with CI_LAST < CI_RCPT"
 #endif
 #if CI_LAST < CI_MAIL
-# ERROR "do not compile with CI_LAST < CI_MAIL"
+# error "do not compile with CI_LAST < CI_MAIL"
 #endif
 #if CI_LAST < CI_HELO
-# ERROR "do not compile with CI_LAST < CI_HELO"
+# error "do not compile with CI_LAST < CI_HELO"
 #endif
 #if CI_LAST < CI_CONN
-# ERROR "do not compile with CI_LAST < CI_CONN"
+# error "do not compile with CI_LAST < CI_CONN"
 #endif
 #if CI_LAST >= MAX_MACROS_ENTRIES
-# ERROR "do not compile with CI_LAST >= MAX_MACROS_ENTRIES"
+# error "do not compile with CI_LAST >= MAX_MACROS_ENTRIES"
 #endif
 
 /* function prototypes */
diff -r 912b8d5daca2 -r 7ce8b17603d2 libsm/ldap.c
--- a/libsm/ldap.c	Thu Mar 09 08:32:49 2023 -0800
+++ b/libsm/ldap.c	Thu Mar 09 08:40:07 2023 -0800
@@ -49,10 +49,10 @@
 
 # if _FFR_LDAP_VERSION
 #  if defined(LDAP_VERSION_MAX) && _FFR_LDAP_VERSION > LDAP_VERSION_MAX
-#   ERROR "_FFR_LDAP_VERSION > LDAP_VERSION_MAX"
+#   error "_FFR_LDAP_VERSION > LDAP_VERSION_MAX"
 #  endif
 #  if defined(LDAP_VERSION_MIN) && _FFR_LDAP_VERSION < LDAP_VERSION_MIN
-#   ERROR "_FFR_LDAP_VERSION < LDAP_VERSION_MAX"
+#   error "_FFR_LDAP_VERSION < LDAP_VERSION_MAX"
 #  endif
 #  define SM_LDAP_VERSION_DEFAULT	_FFR_LDAP_VERSION
 # else /* _FFR_LDAP_VERSION */
@@ -536,7 +536,7 @@
 		if (lmap->ldap_multi_args)
 		{
 # if SM_LDAP_ARGS < 10
-#  ERROR _SM_LDAP_ARGS must be 10
+#  error _SM_LDAP_ARGS must be 10
 # endif
 			if (q[1] == 's')
 				key = argv[0];
diff -r 912b8d5daca2 -r 7ce8b17603d2 mail.local/mail.local.c
--- a/mail.local/mail.local.c	Thu Mar 09 08:32:49 2023 -0800
+++ b/mail.local/mail.local.c	Thu Mar 09 08:40:07 2023 -0800
@@ -1686,7 +1686,7 @@
 	MD5_CTX ctx;
 	unsigned char md5[18];
 #  if MAXPATHLEN <= 24
-#    ERROR "MAXPATHLEN <= 24"
+#    error "MAXPATHLEN <= 24"
 #  endif
 	char b64[24];
 	MD5_LONG bits;
diff -r 912b8d5daca2 -r 7ce8b17603d2 sendmail/conf.c
--- a/sendmail/conf.c	Thu Mar 09 08:32:49 2023 -0800
+++ b/sendmail/conf.c	Thu Mar 09 08:40:07 2023 -0800
@@ -5885,14 +5885,14 @@
 	/* set LDAP_OPT_NETWORK_TIMEOUT if available (-c) */
 	"LDAP_NETWORK_TIMEOUT",
 # else
-#  ERROR "LDAP_NETWORK_TIMEOUT requires LDAPMAP"
+#  error "LDAP_NETWORK_TIMEOUT requires LDAPMAP"
 # endif
 #endif
 #if LDAP_REFERRALS
 # if LDAPMAP
 	"LDAP_REFERRALS",
 # else
-#  ERROR "LDAP_REFERRALS requires LDAPMAP"
+#  error "LDAP_REFERRALS requires LDAPMAP"
 # endif
 #endif
 #if LOG
@@ -5958,7 +5958,7 @@
 #if NEWDB
 # if defined(DB_VERSION_MAJOR) && defined(DB_VERSION_MINOR)
 #  if DB_VERSION_MAJOR >= 5 && !defined(SOLARIS) && !HASFLOCK
-#    ERROR "Berkeley DB file locking needs flock() for version 5.x (and greater?)"
+#    error "Berkeley DB file locking needs flock() for version 5.x (and greater?)"
 #  endif
 	"NEWDB=" SM_XSTR(DB_VERSION_MAJOR) "." SM_XSTR(DB_VERSION_MINOR),
 # else
@@ -6039,10 +6039,10 @@
 	*/
 
 # if !ALLOW_255
-#  ERROR "USE_EAI requires ALLOW_255"
+#  error "USE_EAI requires ALLOW_255"
 # endif
 # if _FFR_EIGHT_BIT_ADDR_OK
-#  ERROR "Cannot enable both USE_EAI and _FFR_EIGHT_BIT_ADDR_OK"
+#  error "Cannot enable both USE_EAI and _FFR_EIGHT_BIT_ADDR_OK"
 # endif
 	"USE_EAI",
 #endif
@@ -6625,13 +6625,13 @@
 #endif
 #if _FFR_MTA_STS
 # if !MAP_REGEX
-#  ERROR "_FFR_MTA_STS requires MAP_REGEX"
+#  error "_FFR_MTA_STS requires MAP_REGEX"
 # endif
 # if !STARTTLS
-#  ERROR "_FFR_MTA_STS requires STARTTLS"
+#  error "_FFR_MTA_STS requires STARTTLS"
 # endif
 # if !_FFR_TLS_ALTNAMES
-#  ERROR "_FFR_MTA_STS requires _FFR_TLS_ALTNAMES"
+#  error "_FFR_MTA_STS requires _FFR_TLS_ALTNAMES"
 # endif
 	/* MTA STS support */
 	"_FFR_MTA_STS",
@@ -6673,7 +6673,7 @@
 	/* outgoing connection control (not yet working) */
 	"_FFR_OCC",
 # else
-#  ERROR "_FFR_OCC requires SM_CONF_SHM"
+#  error "_FFR_OCC requires SM_CONF_SHM"
 # endif
 #endif
 #if _FFR_PROXY
@@ -6772,7 +6772,7 @@
 	/* LDAP debugging */
 	"_FFR_SM_LDAP_DBG",
 # else
-#  ERROR "_FFR_SM_LDAP_DBG requires LDAPMAP and LBER_OPT_LOG_PRINT_FN"
+#  error "_FFR_SM_LDAP_DBG requires LDAPMAP and LBER_OPT_LOG_PRINT_FN"
 # endif
 #endif
 #if _FFR_SPT_ALIGN
@@ -6852,7 +6852,7 @@
 # if defined(X509_V_FLAG_TRUSTED_FIRST)
 	"_FFR_VRFY_TRUSTED_FIRST",
 # else
-#  ERROR "_FFR_VRFY_TRUSTED_FIRST set but X509_V_FLAG_TRUSTED_FIRST not defined"
+#  error "_FFR_VRFY_TRUSTED_FIRST set but X509_V_FLAG_TRUSTED_FIRST not defined"
 # endif
 #endif
 
diff -r 912b8d5daca2 -r 7ce8b17603d2 sendmail/conf.h
--- a/sendmail/conf.h	Thu Mar 09 08:32:49 2023 -0800
+++ b/sendmail/conf.h	Thu Mar 09 08:40:07 2023 -0800
@@ -132,7 +132,7 @@
 	/* must be less than BITMAPBITS for DoQueueRun */
 #endif
 #if MAXQUEUEGROUPS >= BITMAPBITS
-# ERROR "MAXQUEUEGROUPS must be less than BITMAPBITS"
+# error "MAXQUEUEGROUPS must be less than BITMAPBITS"
 #endif
 
 #ifndef MAXWORKGROUPS
diff -r 912b8d5daca2 -r 7ce8b17603d2 sendmail/headers.c
--- a/sendmail/headers.c	Thu Mar 09 08:32:49 2023 -0800
+++ b/sendmail/headers.c	Thu Mar 09 08:40:07 2023 -0800
@@ -1130,11 +1130,11 @@
 #  define FIRSTLOGLEN 850
 # else
 #  if MSGIDLOGLEN < 100
-#    ERROR "MSGIDLOGLEN too short"
+#    error "MSGIDLOGLEN too short"
 #  endif
 /* XREF: this is "sizeof(sbuf)", see above */
 #  if MSGIDLOGLEN >= MAXLINE / 2
-#    ERROR "MSGIDLOGLEN too long"
+#    error "MSGIDLOGLEN too long"
 #  endif
 
 /* 850 - 100 for original MSGIDLOGLEN */
@@ -1142,7 +1142,7 @@
 
 /* check that total length is ok */
 #  if FIRSTLOGLEN + 200 >= MAXLINE
-#    ERROR "MSGIDLOGLEN too long"
+#    error "MSGIDLOGLEN too long"
 #  endif
 #  if MSGIDLOGLEN > MAXNAME
 #    undef XBUFLEN
@@ -1371,7 +1371,7 @@
 	} while (0)
 
 #if MAXNAME < 10
-# ERROR "MAXNAME must be at least 10"
+# error "MAXNAME must be at least 10"
 #endif
 
 char *
diff -r 912b8d5daca2 -r 7ce8b17603d2 sendmail/map.c
--- a/sendmail/map.c	Thu Mar 09 08:32:49 2023 -0800
+++ b/sendmail/map.c	Thu Mar 09 08:40:07 2023 -0800
@@ -4037,7 +4037,7 @@
 	char *argv[SM_LDAP_ARGS];
 	char keybuf[MAXKEY];
 # if SM_LDAP_ARGS != MAX_MAP_ARGS
-#  ERROR "SM_LDAP_ARGS must be the same as MAX_MAP_ARGS"
+#  error "SM_LDAP_ARGS must be the same as MAX_MAP_ARGS"
 # endif
 
 # define AV_FREE(av)	\
diff -r 912b8d5daca2 -r 7ce8b17603d2 sendmail/mci.c
--- a/sendmail/mci.c	Thu Mar 09 08:32:49 2023 -0800
+++ b/sendmail/mci.c	Thu Mar 09 08:40:07 2023 -0800
@@ -1200,7 +1200,7 @@
 		struct dirent *e;
 		char newpath[MAXPATHLEN];
 #if MAXPATHLEN <= MAXNAMLEN - 3
-# ERROR "MAXPATHLEN <= MAXNAMLEN - 3"
+# error "MAXPATHLEN <= MAXNAMLEN - 3"
 #endif
 
 		if ((d = opendir(pathname)) == NULL)
diff -r 912b8d5daca2 -r 7ce8b17603d2 sendmail/ratectrl.c
--- a/sendmail/ratectrl.c	Thu Mar 09 08:32:49 2023 -0800
+++ b/sendmail/ratectrl.c	Thu Mar 09 08:40:07 2023 -0800
@@ -228,7 +228,7 @@
 		}
 	}
 #else /* HASH_ALG == 1 */
-# ERROR "unsupported HASH_ALG"
+# error "unsupported HASH_ALG"
 	hv = ((hv << 1) ^ (*p & 0377)) % cctx->cc_size; ???
 #endif /* HASH_ALG == 1 */
 
diff -r 912b8d5daca2 -r 7ce8b17603d2 sendmail/readcf.c
--- a/sendmail/readcf.c	Thu Mar 09 08:32:49 2023 -0800
+++ b/sendmail/readcf.c	Thu Mar 09 08:40:07 2023 -0800
@@ -2973,7 +2973,7 @@
 #endif
 #if _FFR_EIGHT_BIT_ADDR_OK
 # if !ALLOW_255
-#  ERROR "_FFR_EIGHT_BIT_ADDR_OK requires ALLOW_255"
+#  error "_FFR_EIGHT_BIT_ADDR_OK requires ALLOW_255"
 # endif
 # define O_EIGHT_BIT_ADDR_OK	0xdf
 	{ "EightBitAddrOK",	O_EIGHT_BIT_ADDR_OK,	OI_NONE },
diff -r 912b8d5daca2 -r 7ce8b17603d2 sendmail/sendmail.h
--- a/sendmail/sendmail.h	Thu Mar 09 08:32:49 2023 -0800
+++ b/sendmail/sendmail.h	Thu Mar 09 08:40:07 2023 -0800
@@ -46,16 +46,16 @@
 # endif
 #else /* STARTTLS */
 # if DANE
-#  ERROR "DANE set but STARTTLS not defined"
+#  error "DANE set but STARTTLS not defined"
 # endif
 # if _FFR_TLS_ALTNAMES
-#  ERROR "_FFR_TLS_ALTNAMES set but STARTTLS not defined"
+#  error "_FFR_TLS_ALTNAMES set but STARTTLS not defined"
 # endif
 # if _FFR_TLSFB2CLEAR
-#   ERROR "_FFR_TLSFB2CLEAR set but STARTTLS not defined"
+#   error "_FFR_TLSFB2CLEAR set but STARTTLS not defined"
 # endif
 # if _FFR_TLS_USE_CERTIFICATE_CHAIN_FILE
-#  ERROR "_FFR_TLS_USE_CERTIFICATE_CHAIN_FILE set but STARTTLS not defined"
+#  error "_FFR_TLS_USE_CERTIFICATE_CHAIN_FILE set but STARTTLS not defined"
 # endif
 #endif /* STARTTLS */
 
@@ -140,29 +140,29 @@
 # define ALLOW_255 1
 #endif
 #if _FFR_EAI && _FFR_EIGHT_BIT_ADDR_OK
-# ERROR "Cannot enable both of these FFRs: _FFR_EAI _FFR_EIGHT_BIT_ADDR_OK"
+# error "Cannot enable both of these FFRs: _FFR_EAI _FFR_EIGHT_BIT_ADDR_OK"
 #endif
 
 #if _FFR_OCC && !SM_CONF_SHM
-# ERROR "_FFR_OCC requires SM_CONF_SHM"
+# error "_FFR_OCC requires SM_CONF_SHM"
 #endif
 
 #if _FFR_SM_LDAP_DBG && !(LDAPMAP && defined(LBER_OPT_LOG_PRINT_FN))
-# ERROR "_FFR_SM_LDAP_DBG requires LDAPMAP and LBER_OPT_LOG_PRINT_FN"
+# error "_FFR_SM_LDAP_DBG requires LDAPMAP and LBER_OPT_LOG_PRINT_FN"
 #endif
 
 #if _FFR_LOG_MORE1 > 1 || _FFR_LOG_MORE2 > 1
 # if _FFR_LOG_MORE1 != _FFR_LOG_MORE2
-#  ERROR "_FFR_LOG_MORE1 != _FFR_LOG_MORE2"
+#  error "_FFR_LOG_MORE1 != _FFR_LOG_MORE2"
 # endif
 #endif
 
 #if LDAP_NETWORK_TIMEOUT && !(LDAPMAP && defined(LDAP_OPT_NETWORK_TIMEOUT))
-# ERROR "LDAP_NETWORK_TIMEOUT requires LDAPMAP"
+# error "LDAP_NETWORK_TIMEOUT requires LDAPMAP"
 #endif
 
 #if _FFR_VRFY_TRUSTED_FIRST && !defined(X509_V_FLAG_TRUSTED_FIRST)
-# ERROR "_FFR_VRFY_TRUSTED_FIRST set but X509_V_FLAG_TRUSTED_FIRST not defined"
+# error "_FFR_VRFY_TRUSTED_FIRST set but X509_V_FLAG_TRUSTED_FIRST not defined"
 #endif
 
 #if _FFR_8BITENVADDR
@@ -265,14 +265,14 @@
 #   define SASL SASL_VERSION
 #  else /* SASL == 1 || SASL == 2 */
 #   if SASL != SASL_VERSION
-#    ERROR "README: -DSASL (SASL) does not agree with the version of the CYRUS_SASL library (SASL_VERSION)"
-#    ERROR "README: see README!"
+#    error "README: -DSASL (SASL) does not agree with the version of the CYRUS_SASL library (SASL_VERSION)"
+#    error "README: see README!"
 #   endif /* SASL != SASL_VERSION */
 #  endif /* SASL == 1 || SASL == 2 */
 # else /* defined(SASL_VERSION_MAJOR) && defined(SASL_VERSION_MINOR) && defined(SASL_VERSION_STEP) */
 #  if SASL == 1
-#   ERROR "README: please set -DSASL to the version of the CYRUS_SASL library"
-#   ERROR "README: see README!"
+#   error "README: please set -DSASL to the version of the CYRUS_SASL library"
+#   error "README: see README!"
 #  endif /* SASL == 1 */
 # endif /* defined(SASL_VERSION_MAJOR) && defined(SASL_VERSION_MINOR) && defined(SASL_VERSION_STEP) */
 #endif /* SASL */
@@ -767,7 +767,7 @@
 	(SASL_SEC_FORWARD_SECRECY & SASL_SEC_MASK) == 0 || \
 	(SASL_SEC_NOANONYMOUS & SASL_SEC_MASK) == 0 || \
 	(SASL_SEC_PASS_CREDENTIALS & SASL_SEC_MASK) == 0
-#   ERROR "change SASL_SEC_MASK notify sendmail.org!"
+#   error "change SASL_SEC_MASK notify sendmail.org!"
 #  endif /* SASL_SEC_NOPLAINTEXT & SASL_SEC_MASK) == 0 ... */
 # endif /* SASL >= 20101 */
 # define MAXOUTLEN 8192	/* length of output buffer, should be 2^n */
diff -r 912b8d5daca2 -r 7ce8b17603d2 sendmail/srvrsmtp.c
--- a/sendmail/srvrsmtp.c	Thu Mar 09 08:32:49 2023 -0800
+++ b/sendmail/srvrsmtp.c	Thu Mar 09 08:40:07 2023 -0800
@@ -579,7 +579,7 @@
 # define MAXSHIFT 8
 #endif
 #if MAXSHIFT > 31
-# ERROR "MAXSHIFT > 31 is invalid"
+# error "MAXSHIFT > 31 is invalid"
 #endif
 
 
@@ -874,7 +874,7 @@
 	char *args[MAXSMTPARGS];
 	char inp[MAXINPLINE];
 #if MAXINPLINE < MAXLINE
-# ERROR "MAXINPLINE must NOT be less than MAXLINE"
+# error "MAXINPLINE must NOT be less than MAXLINE"
 #endif
 	char cmdbuf[MAXLINE];
 #if SASL
@@ -5748,7 +5748,7 @@
 	       sasl_external_properties_t * ext_ssf)
 # endif /* SASL >= 20000 */
 #else /* __STDC__ */
-# ERROR "SASL requires __STDC__"
+# error "SASL requires __STDC__"
 #endif /* __STDC__ */
 {
 	int result;
diff -r 912b8d5daca2 -r 7ce8b17603d2 sendmail/tls.c
--- a/sendmail/tls.c	Thu Mar 09 08:32:49 2023 -0800
+++ b/sendmail/tls.c	Thu Mar 09 08:40:07 2023 -0800
@@ -26,7 +26,7 @@
 # include <tls.h>
 
 # if defined(OPENSSL_VERSION_NUMBER) && OPENSSL_VERSION_NUMBER <= 0x00907000L
-#  ERROR "OpenSSL version OPENSSL_VERSION_NUMBER is unsupported."
+#  error "OpenSSL version OPENSSL_VERSION_NUMBER is unsupported."
 # endif
 
 /*
